carta/session.py

```diff
@@ -358,39 +359,58 @@ def set_starting_directory(self, path):
 
     # IMAGES
 
-    @validate(String(), String(r"\d*"), Boolean(), Constant(ArithmeticExpression))
-    def open_image(self, path, hdu="", complex=False, expression=ArithmeticExpression.AMPLITUDE):
-        """Open a new image, replacing any existing images.
+    @validate(String(), String(r"\d*"), Boolean())
+    def open_image(self, path, hdu="", append=False):
+        """Open or append a new image.
 
         Parameters
         ----------
         path : {0}
             The path to the image file, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
         hdu : {1}
             The HDU to select inside the file.
-        complex : {2}
-            Whether the image is complex. Set to ``False`` by default.
-        expression : {3}
-            Arithmetic expression to use if opening a complex-valued image. The default is :obj:`carta.constants.ArithmeticExpression.AMPLITUDE`.
+        append : {2}
+            Whether the image should be appended. Default is ``False``.
         """
-        return Image.new(self, path, hdu, False, complex, expression)
+        path = self.resolve_file_path(path)
+        directory, file_name = posixpath.split(path)
+        return Image.new(self, directory, file_name, hdu, append, False)
 
-    @validate(String(), String(r"\d*"), Boolean(), Constant(ArithmeticExpression))
-    def append_image(self, path, hdu="", complex=False, expression=ArithmeticExpression.AMPLITUDE):
-        """Append a new image, keeping any existing images.
+    @validate(String(), Constant(ComplexExpression), Boolean())
+    def open_complex_image(self, path, expression=ComplexExpression.AMPLITUDE, append=False):
+        """Open or append a new complex-valued image.
 
         Parameters
         ----------
         path : {0}
-            The path to the image file, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
-        hdu : {1}
-            The HDU to select inside the file.
-        complex : {2}
-            Whether the image is complex. Set to ``False`` by default.
-        expression : {3}
-            Arithmetic expression to use if appending a complex-valued image. The default is :obj:`carta.constants.ArithmeticExpression.AMPLITUDE`.
-        """
-        return Image.new(self, path, hdu, True, complex, expression)
+            The path to the complex-valued image file, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
+        expression : {1}
+            The complex expression for opening a complex-valued image.
```

Comment:
Please add back:
```
The default is :obj:`carta.constants.ArithmeticExpression.AMPLITUDE`.
```

Reply:
Sorry, one more thing: the first part should read: `The complex expression to use when opening the image.` (The old description came from the more generic function; this function is *only* for complex-valued images.)